server/controllers/documentController.js

```diff
@@ -1,155 +1,163 @@
+    return Document
+      .create({ title, authorId, content, access, roleId: user.roleId })
+      .then((doc) => {
+        const { roleId, ...documentData } = doc.dataValues;
+        return response.status(201).json({
+          document: documentData
+        });
+      }
       )
-      .catch(error => helpers.handleCreateDocumentError(error, response));
+      .catch(error => documentHelpers
+        .handleCreateDocumentError(error, response));
   },
+  /**
+  * @description gets an array of documents available in the database
+  * @param {object} request http request object from express
+  * @param {object} response http response object  from expressjs
+  * @returns {Promise} promise from express http response
+  */
   getDocuments: (request, response) => {
-    const options = {};
-    const { limit, offset } = request.query;
-    const isPaginationRequired = limit && offset;
-    if (isPaginationRequired) {
-      if (Number.isNaN(Number(limit)) || Number.isNaN(Number(offset))) {
-        return response
-          .status(406)
-          .json({ error: errorMessages.paginationQueryError });
-      }
-      options.limit = Number.parseInt(limit, 10);
-      options.offset = Number.parseInt(offset, 10);
-    }
-    options.where = {
-      $or: [{ access: 'public' }, {
-        access: 'role',
-        $and: { role: response.locals.user.role }
-      }, {
-        access: 'private',
-        $and: { author: response.locals.user.id }
-      }]
-    };
+    const paginationQueryStrings = response.locals.paginationQueryStrings;
+    const currentUser = response.locals.user;
+    const options = documentHelpers
+      .generateFindDocumentsOptions(currentUser, paginationQueryStrings);
     return Document.findAndCountAll(options)
       .then((docs) => {
         let statusCode = 200;
         const responseData = { documents: docs.rows, count: docs.count };
-        if (isPaginationRequired) {
+        if (paginationQueryStrings) {
           const pageMetadata = getPageMetadata(
             options.limit,
             options.offset,
             docs);
           if (!docs.rows[0]) {
-            pageMetadata.message = errorMessages.endOfPageReached;
+            pageMetadata.message = errorConstants.endOfPageReached;
             statusCode = 404;
           }
           responseData.pageMetadata = pageMetadata;
         } else if (!docs.rows[0]) {
           return response
-            .status(404).json({ error: errorMessages.noDocumentFoundError });
+            .status(404).json({ error: errorConstants.noDocumentFoundError });
         }
         return response.status(statusCode).json(responseData);
       });
   },
+  /**
+   * @description gets one document from database
+   * @param {object} request expressjs request object
+   * @param {object} response  expressjs response object
+   * @returns {Promise} promise from express http resonse object
+   */
   getDocument(request, response) {
```

Comment:
Why not follow the same method of declaring methods you adopted for other methods in the class?
getDocument: () => {}
getDocument() => {}

Reply:
okay. I would try to correct this